lib/sensors/imd.cpp

```diff
@@ -0,0 +1,66 @@
+#include <imd.hpp>
+
+namespace hyped::sensors {
+std::optional<std::shared_ptr<Imd>> Imd::create(core::ILogger &logger,
+                                                const std::shared_ptr<io::ICan> can)
+{
+  std::optional<std::shared_ptr<Imd>> optional_imd = std::make_shared<Imd>(logger, can);
+  if (!optional_imd) {
+    logger.log(core::LogLevel::kFatal, "Failed to create IMD");
+    return std::nullopt;
+  }
+  std::shared_ptr<Imd> imd = *optional_imd;
+  can.addProcessor(kReturnDataCanId, imd);
```

Comment:
also, i have messed up the typing on addProcessor, `id` should be a `uint32_t`. i forgot that extended can used longer ids. if you update hardware_can.[ch]pp and can.hpp then this will work.